irods/rule.py

```diff
@@ -65,8 +66,10 @@ def load(self, rule_file, encoding = 'utf-8'):
         self.body = '@external\n'
 
         # parse rule file
-        with io_open(rule_file, encoding = encoding) as f:
+        with (io_open(rule_file, encoding = encoding) if isinstance(rule_file,six.string_types) else rule_file
+        ) as f:
             for line in f:
+                if isinstance(f, bytes): f = f.decode(encoding)
```

Comment:
consider the two-line, indented formulation, and a comment explaining the 'why'

Reply:
Done!